test/libsolidity/semanticTests/abiEncoderV2/calldata_struct_dynamic_and_static.sol

```diff
@@ -0,0 +1,17 @@
+pragma abicoder               v2;
+
+contract C {
+    function f(uint256[] memory a, uint256[1] calldata b) public returns (bytes memory) {
+        return abi.encode(a, b);
+    }
+
+    function g(uint256[] memory a, uint256[1] calldata b) external returns (bytes memory) {
+        return f(a, b);
+    }
+}
```

Comment:
I think we should cover more situations.

And we should also make sure that every use of `copyToMemoryFunction()` is covered by some test.

Reply:
> And we should also make sure that every use of copyToMemoryFunction() is covered by some test.

Why, if this case is the only one that uses it without cleanup?
The rest wasn't changed..

Reply:
I investigated the other usages of memcopy and couldn't find any scenario where its cleaning might cause problems.

We could add more tests that use different types of dyn and static arrays, but other than that I can't think of more variations to test here...

And I doubt they add much, either they take the same codepath and use the fixed memcopy or they use a completely different encoding and thus we're not testing this problem..

Reply:
I just pushed a commit that adds some of the more important tests from the list above: for errors, events and library functions. Also the more complex case from the blog post.

Also added some small tweaks to existing tests - renaming (they no longer contain structs), no homestead and abiv2 required (should still pass in all of these cases). Also added a function that returns the parameters directly, without explicit encoding.

Left the modifications unsquashed so that they're easy to review.